simularium/simularium-website
# Feature/looping playback

Problem
=======
Gives users the ability to loop simulation playback, as well as setting primary global styling for tooltips.

Closes [#298](https://github.com/simularium/simularium-website/issues/298)

Solution
========
* New feature (non-breaking change which adds functionality)

Looping button was added to UI, clicking it toggles a new boolean piece of state in the viewer branch. Playback functions, icons, and tooltips respond to that state.

Some styles were removed and some added to make tooltips all have the same look.

Steps to Verify:
----------------
Load a simulation, find and click the looping icon in the playback controls, run a simulation and see if its starts over when it reaches the last frame.

Check styling on tooltips by hovering cursor over left panel checkboxes, playback controls, and home icon. Should match [design](https://xd.adobe.com/view/75b6a3c8-f650-443b-bccc-d1fcd134a35e-20e5/screen/324d327c-ade5-4765-8c66-aa61db1feb4f/specs/).

Keyfiles (delete if not relevant):
-----------------------
1. UI elements added to PlaybackControls/index.tsx
2. Looping state changes the standard suite of Redux files.
3. Function to toggle looping state is is ViewerPanel/index.tsx, as is function receiveTimeChange which was altered with logic for looping playback.
